### PR TITLE
Haiku supporting patches

### PR DESCRIPTION
**Pull request by extrowerk**
### This pullrequest changes
Basic Haiku OS support

Recipe already here:
https://github.com/haikuports/haikuports/pull/2281

### CODE REVIEW

File: cmake/OpenCVCompilerDefenses.cmake [hunk 1]
```diff
@@ -70,7 +70,11 @@ else()
   # not supported
 endif()
 
-set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+if(NOT HAIKU)
+  set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+else()
+  set(CMAKE_POSITION_INDEPENDENT_CODE FALSE)
```
[the remark this answers is not in the document]
Comment (extrowerk) [reply cmake/OpenCVCompilerDefenses.cmake]: OH, sorry, my mistake. Just grepped for hardening and guarded with ifdef, not checked if it is processed by default or not. Will remove this.

[the remark this answers is not in the document]
Comment (alalek) [reply cmake/OpenCVCompilerDefenses.cmake]: ping

**Comment by extrowerk** [comment]
Pong. Vacation...

-- Z.

2018.03.12. dátummal, 19:50 időpontban Alexander Alekhin <notifications@github.com> írta:

**Comment by extrowerk** [comment]
Updated.

**Pull request merged**